web/app/org/ada/web/controllers/dataset/DataSetController.scala

```diff
@@ -158,6 +158,16 @@ trait DataSetController extends ReadonlyController[BSONObjectID] {
     filterOrId: FilterOrId
   ): Action[AnyContent]
 
+  def getLineChart(
+    filterOrId: FilterOrId
+  ): Action[AnyContent]
+
+  def calcLineChart(
+    xFieldName: String,
+    groupFieldName: Option[String],
+    filterOrId: FilterOrId
+  ): Action[AnyContent]
+
```

Comment:
Those seem awfully specific for something as generic as a "DataSetController". You're sure this is a good place for them?

Reply:
yes it's as general as possible for a line chart